src/targets/gpu/target.cpp

```diff
@@ -161,14 +162,20 @@ std::vector<pass> target::get_passes(migraphx::context& gctx, const compile_opti
         dead_code_elimination{},
         normalize_ops{},
         dead_code_elimination{},
+        // 1. For int4 ops: first remove the identity op next to packed int4 data.
+        eliminate_identity{},
+        // 2. Next: (pack/unpack)_int4 handling in simplyfy_qdq.
         simplify_qdq{},
+        // 3. Next: pruning of pack_int4 related const branches.
+        propagate_constant{{}},
```

Comment:
I think this should be removed, its not necessary for int4-weights, and I am concerned it might cause perf regressions.

Reply:
It has been removed. However, I didn't notice any perf regression below.